### DISCUSSION TITLE
Question: Can I change the copy-to-clipboard destination URL of an upload?

### DISCUSSION THREAD

**Discussion by andrewjroberts** [payloadcms/payload#1278] [Q&A]
Hi all, 

Can I change the copy-to-clipboard destination URL of an upload?

My team has a [Cloudinary plugin](https://github.com/richardvanbergen/payload-plugin-cloud-storage/tree/feature/cloudinary) configured for our Media strategy, but after digging around `IncomingUploadType` properties I can't figure out if there's a way to set the copy button of the "file-details" UI component for Uploads.

The Admin Thumbnail works fine, i.e.
`condition: (data) => Boolean(data?.cloudinaryURL)`

... but we need something similar for the copy button (or I'll just hide with CSS if not possible).

<img width="455" alt="Screen Shot 2022-10-19 at 11 16 04 AM" src="https://user-images.githubusercontent.com/101012384/196734183-2a41fd61-f5f4-44c6-b06c-827d70c79e89.png">

- in the above screenshot, you can set the root path of the URL & its dir, but what should I do if it's Cloudinary with a hashed root?

**Comment by jmikrut** [comment]
Hey @andwrobs — it looks like you're using the unofficial cloud storage plugin, which _**adds an additional URL field**_ instead of properly overwriting the existing `url` field that comes with Payload uploads. We have an official Cloud Storage Plugin that functions in mostly the same way, although is written to automatically update file URLs so that they point to your cloud storage directly. That means that the "copy to clipboard" URLs will automatically point to the correct destination as a result.

If you want to do this and keep your existing implementation with the unofficial plugin, I'd suggest an `afterRead` hook that updates the built-in `url` field with your `cloudinaryURL` value. That way this will all sort itself out. Otherwise you could adapt your Cloudinary adapter to work with our official plugin and then this will all be handled for you.

What do you think?

**Reply by andrewjroberts** [reply]
@jmikrut thanks for the quick response & clear explanation as always 🙏 

I did not know the official cloud storage plugin handled that! Let me see if I can rejig @richardvanbergen's unofficial Cloudinary plugin to conform to the Adapter interface, may ask some questions/for some validation.

**Reply by richardvanbergen** [reply]
@andwrobs I'll be honest with you that project has kind of fallen by the wayside as my current position doesn't call for a CMS. If there's anything I can do to help let me know but there's probably been a lot of developments since I published my plugin.

**Reply by andrewjroberts** [reply]
@richardvanbergen thank you so much for your work on it & the offer for help! I personally ended up going with the official cloud storage plugin—but someone should very easily be able to port your plugin to the new Adapter interface as it's just a light wrapper on top how you already split up the handler methods. cheers man!
